### PR TITLE
[Core] expose quaternion function to python

### PR DESCRIPTION
**Pull request by VeronikaSinger**
Expose quaternion functions to python.
Is needed for #7176

### CODE REVIEW

File: kratos/python/add_quaternion_to_python.cpp [hunk 1]
```diff
@@ -48,6 +49,10 @@ void  AddQuaternionToPython(pybind11::module& m) {
     .def_property("Y", QuaternionGetY, QuaternionSetY)
     .def_property("Z", QuaternionGetZ, QuaternionSetZ)
     .def_property("W", QuaternionGetW, QuaternionSetW)
+    .def("Normalize",        &Quaternion<double>::normalize)
+    .def_static("Identity",  &Quaternion<double>::Identity)
+    .def("ToRotationMatrix", &Quaternion<double>::ToRotationMatrix<Matrix>)
+    .def("MultiplyQuaternion", MultiplyQuaternion)
```
Comment (philbucher) [inline kratos/python/add_quaternion_to_python.cpp]: If you use a lambda then you don't need the aux function

```suggestion
    .def("MultiplyQuaternion", [](Quaternion<double>& self, Quaternion<double>& a, Quaternion<double>& b){return Quaternion<double>(operator*(a,b));})
```

BTW why does this have three arguments and not two? Can you show the usage?

**Review by philbucher — COMMENTED** [review]
looks good to me
(tests were already added in #6769)

File: kratos/python/add_quaternion_to_python.cpp (hunk 1 above)
Comment (VeronikaSinger) [reply kratos/python/add_quaternion_to_python.cpp]: 3 arguments are needed as 2 input quaternions for the multiplication and the first argument is the quaternion which calls the function. I'm calling the function like that in my code:
self._quaternion = self._quaternion.MultiplyQuaternion(self._delta_quaternion, self._quaternion)

Comment (philbucher) [reply kratos/python/add_quaternion_to_python.cpp]: ok I understand
I checked the c++ code it is consistent

**Review by philbucher — APPROVED** [review]
:+1:

please wait with merging until Friday in case someone else has a comment

**Comment by VeronikaSinger** [comment]
OK
Thanks @philbucher

**Comment by maceligueta** [comment]
No comments from my side. Good additions to improve usability from Python.
Thanks @VeronikaSinger  and @philbucher

**Pull request merged**